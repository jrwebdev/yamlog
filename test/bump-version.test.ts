it('should equal true', () => {
  expect(true).toEqual(false);
});
