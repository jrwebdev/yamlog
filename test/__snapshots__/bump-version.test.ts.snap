// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`package.json should write the version to package.json 1`] = `
1.0.1:
  metadata:
    date: '2017-05-20'
  fix:
    - Fix 1
1.0.0:
  metadata:
    date: '2017-01-01'
  feature:
    - Feature 1

`;

exports[`should bump a major version if there are unreleased breaking changes 1`] = `
2.0.0:
  metadata:
    date: '2017-05-20'
  breaking:
    - Breaking Change 1
  feature:
    - Feature 1
  fix:
    - Fix 1
1.0.0:
  metadata:
    date: '2017-01-01'
  feature:
    - Feature 1

`;

exports[`should bump a minor version if there are unreleased features 1`] = `
1.1.0:
  metadata:
    date: '2017-05-20'
  feature:
    - Feature 1
  fix:
    - Fix 1
1.0.0:
  metadata:
    date: '2017-01-01'
  feature:
    - Feature 1

`;

exports[`should bump a minor version if there are unreleased fixes for an unstable project 1`] = `
0.2.0:
  metadata:
    date: '2017-05-20'
  breaking:
    - Breaking Change 1
  feature:
    - Feature 1
  fix:
    - Fix 1
0.1.0:
  metadata:
    date: '2017-01-01'
  feature:
    - Feature 1

`;

exports[`should bump a patch version if there are unreleased features for an unstable project 1`] = `
0.1.1:
  metadata:
    date: '2017-05-20'
  feature:
    - Feature 1
  fix:
    - Fix 1
0.1.0:
  metadata:
    date: '2017-01-01'
  feature:
    - Feature 1

`;

exports[`should bump a patch version if there are unreleased fixes 1`] = `
1.0.1:
  metadata:
    date: '2017-05-20'
  fix:
    - Fix 1
1.0.0:
  metadata:
    date: '2017-01-01'
  feature:
    - Feature 1

`;

exports[`should bump a patch version if there are unreleased fixes for an unstable project 1`] = `
0.1.1:
  metadata:
    date: '2017-05-20'
  fix:
    - Fix 1
0.1.0:
  metadata:
    date: '2017-01-01'
  feature:
    - Feature 1

`;

exports[`should bump to v1 regardless of change type if an unstable project is now stable 1`] = `
1.0.0:
  metadata:
    date: '2017-05-20'
  fix:
    - Fix 1
0.1.0:
  metadata:
    date: '2017-01-01'
  feature:
    - Feature 1

`;

exports[`should ignore the unstable option if the project is already at v1 1`] = `
2.0.0:
  metadata:
    date: '2017-05-20'
  breaking:
    - Breaking Change 1
1.0.0:
  metadata:
    date: '2017-01-01'
  feature:
    - Feature 1

`;

exports[`should leave the changelog unchanged if there are no unreleased changes 1`] = `
1.0.0:
  metadata:
    date: '2017-01-01'
  feature:
    - Feature 1

`;

exports[`unreleased directory should bump a breaking version if there is a breaking file in the unreleased directory 1`] = `
2.0.0:
  metadata:
    date: '2017-05-20'
  breaking:
    - Breaking Change 1
    - Breaking Change 2
  feature:
    - Feature 1
  fix:
    - Fix 1
1.0.0:
  metadata:
    date: '2017-01-01'
  feature:
    - Feature 1

`;

exports[`unreleased directory should bump a minor version if there is a feature file in the unreleased directory 1`] = `
1.1.0:
  metadata:
    date: '2017-05-20'
  feature:
    - Feature 1
    - Feature 2
    - Feature 3
  fix:
    - Fix 1
1.0.0:
  metadata:
    date: '2017-01-01'
  feature:
    - Feature 1

`;

exports[`unreleased directory should bump a patch version if there is a fix file in the unreleased directory 1`] = `
1.0.1:
  metadata:
    date: '2017-05-20'
  fix:
    - Fix 1
    - Fix 2
1.0.0:
  metadata:
    date: '2017-01-01'
  feature:
    - Feature 1

`;
